server/src/main/java/com/flipkart/varadhi/VerticleDeployer.java

```diff
@@ -104,10 +109,46 @@ private List<RouteDefinition> getDefinitions() {
                 .collect(Collectors.toList());
     }
 
+    private List<RouteDefinition> getAuthZServerDefinitions() {
+        return Stream.of(
+                        defaultAuthZHandlers.get(),
+                        healthCheckHandler.get()
+                )
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
+    }
+
 
     public void deployVerticle(
             Vertx vertx,
             ServerConfiguration configuration
+    ) {
+        if (configuration.isDefaultAuthorizationServerEnabled()) {
+            deployDefaultAuthZVerticle(vertx, configuration);
```

Comment:
deploying as a separate verticle will not cause issues with roundrobin

Reply:
have tested locally

Reply:
mainly thinking from pov of isolation
if not needed then we can have it as part of single router